apps/cloud_inference/server_app/source/inference_hip_rocal.cpp

```diff
@@ -0,0 +1,1092 @@
+               std::get<3>(info) == dimInput[2] &&
+               std::get<4>(info) == dimOutput[0] &&
+               std::get<5>(info) == dimOutput[1] &&
+               std::get<6>(info) == dimOutput[2])
+            {
+                reverseInputChannelOrder = std::get<7>(info);
+                preprocessMpy[0] = std::get<8>(info);
+                preprocessMpy[1] = std::get<9>(info);
+                preprocessMpy[2] = std::get<10>(info);
+                preprocessAdd[0] = std::get<11>(info);
+                preprocessAdd[1] = std::get<12>(info);
+                preprocessAdd[2] = std::get<13>(info);
+                modelPath = args->getConfigurationDir() + "/" + modelName;
+                found = true;
+                break;
+            }
+        }
+    }
+    if(found) {
+        modulePath = modelPath + "/build/" + MODULE_LIBNAME;
+        moduleHandle = dlopen(modulePath.c_str(), RTLD_NOW | RTLD_LOCAL);
+        if(!moduleHandle) {
+            found = false;
+            error("could not locate module %s for %s", modulePath.c_str(), clientName.c_str());
+        }
+        if (args->getModelCompilerPath().empty()) {
+            if(!(annCreateGraph = (type_annCreateGraph *) dlsym(moduleHandle, "annCreateGraph"))) {
+                found = false;
+                error("could not find function annCreateGraph() in module %s for %s", modulePath.c_str(), clientName.c_str());
+            }
+        }
+        else if(!(annAddtoGraph = (type_annAddToGraph *) dlsym(moduleHandle, "annAddToGraph"))) {
+            found = false;
+            error("could not find function annAddToGraph() in module %s for %s", modulePath.c_str(), clientName.c_str());
+        }
+    }
+    else {
+        error("unable to find requested model:%s input:%dx%dx%d output:%dx%dx%d from %s", modelName.c_str(),
+              dimInput[2], dimInput[1], dimInput[0], dimOutput[2], dimOutput[1], dimOutput[0], clientName.c_str());
+    }
+    if(!found) {
+        // send and wait for INFCOM_CMD_DONE message
+        InfComCommand reply = {
+            INFCOM_MAGIC, INFCOM_CMD_DONE, { 0 }, { 0 }
+        };
+        ERRCHK(sendCommand(sock, reply, clientName));
+        ERRCHK(recvCommand(sock, reply, clientName, INFCOM_CMD_DONE));
+        close(sock);
+        return -1;
+    }
+    info("found requested model:%s input:%dx%dx%d output:%dx%dx%d from %s", modelName.c_str(),
+          dimInput[2], dimInput[1], dimInput[0], dimOutput[2], dimOutput[1], dimOutput[0], clientName.c_str());
+
+    // send and wait for INFCOM_CMD_INFERENCE_INITIALIZATION message
+    InfComCommand updateCmd = {
+        INFCOM_MAGIC, INFCOM_CMD_INFERENCE_INITIALIZATION, { 0 }, "started initialization"
+    };
+    ERRCHK(sendCommand(sock, updateCmd, clientName));
+    ERRCHK(recvCommand(sock, updateCmd, clientName, INFCOM_CMD_INFERENCE_INITIALIZATION));
+    info(updateCmd.message);
+
+#if INFERENCE_SCHEDULER_MODE == NO_INFERENCE_SCHEDULER
+#if DONOT_RUN_INFERENCE
```

Comment:
does the rocAL mode work without scheduler? If node we should get rid of the codes without scheduler